gadget/gadget.go

```diff
@@ -400,15 +415,76 @@ func validateVolume(name string, vol *Volume) error {
 			knownFsLabels[s.Label] = true
 		}
 
+		switch s.Role {
+		case SystemSeed:
+			if state.SystemSeed != nil {
+				return fmt.Errorf("cannot have more than one system-seed role")
+			}
+			state.SystemSeed = &vol.Structure[idx]
+		case SystemData:
+			if state.SystemData != nil {
+				return fmt.Errorf("cannot have more than one system-seed role")
+			}
+			state.SystemData = &vol.Structure[idx]
+		}
+
 		previousEnd = end
 	}
 
+	if err := ensureVolumeConsistency(state, constraints); err != nil {
+		return err
+	}
+
 	// sort by starting offset
 	sort.Sort(byStartOffset(structures))
 
 	return validateCrossVolumeStructure(structures, knownStructures)
 }
 
+func ensureVolumeConsistency(state *validationState, constraints *ModelConstraints) error {
+	if constraints == nil {
+		if state.SystemData == nil {
+			return nil
+		}
+
+		// gadget must be auto-consistent if constraints are not specified
+		if state.SystemSeed != nil {
+			if state.SystemData.Label != "" {
```

Comment:
@mvo5 @cmatsuoka   what about the label of SystemSeed itself ?

Reply:
same rule I think

Reply:
Added validation: ensure that the label for the system-seed role is also not defined.